src/Agoda.Analyzers.CodeFixes/AgodaCustom/AG0020FixProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using System;
+using System.Composition;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Agoda.Analyzers.AgodaCustom;
+using Agoda.Analyzers.Helpers;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Agoda.Analyzers.CodeFixes.AgodaCustom
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(AG0020FixProvider))]
+    [Shared]
+    public class AG0020FixProvider : CodeFixProvider
+    {
+        // this isn't very nice, we need to move IDs to a shared const class
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create("AG0020");
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                context.RegisterCodeFix(
+                    CodeAction.Create(
+                        CustomRulesResources.AG0020FixTitle,
+                        cancellationToken => GetTransformedDocumentAsync(context.Document, diagnostic, cancellationToken),
+                        nameof(AG0020FixProvider)),
+                    diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Is it a best practice to have `ConfigureAwait` as part of the fixers?

Reply:
It can never harm to use `.ConfigureAwait(false)`.